app/workers/recover_lost_jobs_worker.rb

```diff
@@ -15,6 +17,23 @@ def old_unprocessed(scope)
     scope.where(processed_at: nil).where("created_at <= ?", 1.hour.ago)
   end
 
+  def non_existent_daily_digests
+    expected_digest_week.map { |date| DigestRun.new(date: date, range: :daily) }
+      .reject { |digest_run| DigestRun.exists?(date: digest_run.date, range: :daily) }
```

Comment:
I wondered the same. I don't think we can use `persisted?`, because it [doesn't actually query the DB](https://apidock.com/rails/ActiveRecord/Persistence/persisted%3F).

Reply:
It doesn't need to query the db as that is already as part of `find_or_initialize_by` the presence of an id column is sufficient to determine the record is new.

Reply:
Aha, I missed `find_or_initialize_by`. That reads a bit better - updated 👍.